File: movies-backend/src/main/java/net/talaatharb/movies/model/Movie.java
package net.talaatharb.movies.model;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Table
@Data
public class Movie {
    @Id
    private Integer id;
    private String title;
    private String overview;
    private String poster_path;
    private String backdrop_path;
    private Integer vote_average;
    private Integer vote_count;
    private boolean adult;
    private String release_date;
    private String original_title;
    private String revenue;
}

File: movies-frontend/src/models/Movie.ts
export interface Movie {
  adult: boolean
  backdrop_path: string
  id: number
  original_title: string
  overview: string
  poster_path: string
  release_date: string
  title: string
  vote_average: number
  vote_count: number
  revenue: string
}

export interface MovieReview {
  author: string
  author_details: {
    name: string
    username: string
    avatar_path: string
    rating: number
  }
  content: string
  created_at: string
  id: string
  updated_at: string
  url: string
}
